### PR TITLE
refactor(yaml): simplify and rename `dropEndingNewline()`

### PR DESCRIPTION
**Pull request by timreichen**
- renames `dropEndingNewline()` to `trimTrailingNewline()`
- uses `.at(-1)`
- adds tests

### CODE REVIEW

File: yaml/_dumper.ts [hunk 1]
```diff
@@ -578,13 +577,11 @@ function writeScalar(
         return `'${string.replace(/'/g, "''")}'`;
       case STYLE_LITERAL:
         return `|${blockHeader(string, state.indent)}${
-          dropEndingNewline(
-            indentString(string, indent),
-          )
+          trimTrailingNewline(indentString(string, indent))
```
Comment (iuioiua) [inline yaml/_dumper.ts:580]: Anything wrong with this?
```suggestion
          indentString(string, indent).trimEnd()
```

Comment (timreichen) [reply yaml/_dumper.ts:580]: I think this will also trim whitespaces. Not sure if that is a problem though.

Comment (iuioiua) [reply yaml/_dumper.ts:585]: Yeah, it'll trim all whitespaces, and I think that's fine. @kt3k, see any issues with that?

Comment (kt3k) [reply yaml/_dumper.ts:585]: That will break the case when the string has white spaces at the end.

(The 2nd case of https://github.com/denoland/deno_std/pull/5341)

**Review by kt3k — APPROVED** [review]

File: yaml/_dumper.ts (hunk 1 above)
Comment (iuioiua) [reply yaml/_dumper.ts:585]: I know, but in what cases is that an issue? Genuine question.

Comment (kt3k) [reply yaml/_dumper.ts:585]: When the user's input has trailing whitespace, then that part needs to be serialized as is. Trimming that whitespace means the user loses some data while serializing.

Comment (iuioiua) [reply yaml/_dumper.ts:585]: Oh, yes, that's right. Obvious once you say it 😆

**Pull request merged**